Source/engine/render/light_render.cpp

```diff
@@ -88,14 +88,28 @@ void RenderTriangle(Point p1, Point p2, Point p3, uint8_t lightLevel, uint8_t *l
 		int cx2 = cy2;
 		int cx3 = cy3;
 
-		for (int x = minx; x < maxx; x++) {
-			if (cx1 > 0 && cx2 > 0 && cx3 > 0)
-				dst[x] = lightLevel;
-
-			cx1 -= fdy12;
-			cx2 -= fdy23;
-			cx3 -= fdy31;
-		}
+		int cxe1 = cx1 - fdy12 * (maxx - minx);
+		int cxe2 = cx2 - fdy23 * (maxx - minx);
+		int cxe3 = cx3 - fdy31 * (maxx - minx);
+
+		int startx = std::max({
+		    cx1 > 0 ? minx : (cxe1 > 0 ? minx + (cx1 / fdy12 + ((cx1 % fdy12) != 0 ? 1 : 0)) : maxx),

```

Comment:
I think there is a faster way to do `(cx1 / fdy12 + ((cx1 % fdy12) != 0 ? 1 : 0)`:

```c++
(cx1 + fdy12 - 1) / fdy12
```

Reply:
This is fine for positive values of `fdy12`. I tweaked it a bit to account for negative numbers as well.